tests/integration/s3/test_s3.py

```diff
@@ -1816,6 +1816,101 @@ def test_s3_copy_object_with_checksum(self, s3_create_bucket, snapshot, aws_clie
         )
         snapshot.match("copy-object-to-dest-keep-checksum", resp)
 
+    @pytest.mark.aws_validated
+    @pytest.mark.skip_snapshot_verify(paths=["$..ServerSideEncryption"])
+    def test_s3_copy_object_preconditions(self, s3_bucket, snapshot, aws_client):
+        snapshot.add_transformer(snapshot.transform.s3_api())
+        object_key = "source-object"
+        dest_key = "dest-object"
+        # create key with no checksum
+        put_object = aws_client.s3.put_object(
+            Bucket=s3_bucket,
+            Key=object_key,
+            Body=b"data",
+        )
+        head_obj = aws_client.s3.head_object(Bucket=s3_bucket, Key=object_key)
+        snapshot.match("head-object", head_obj)
+
+        # wait for the condition to pass
+        time.sleep(3)
```

Comment:
question: which condition exactly? is this some date expiry that is being tested here?

Reply:
right yes, the `if-modified-since` needs a bit of time for the condition to pass, initially I had put 1 sec for LS and 3 for AWS but it failed in the GH actions, I don't like sleeping in test but for those case... S3 compares the `last-modified` field, but you can't set the value in the future otherwise it ignores it. So it needs to be now or less, but the object needs to be a bit earlier than that. I can update the comment while rebasing